### PR TITLE
fix(UVE): RunnableLink construction for page tool analytics #29206

### LINKED ISSUE #29206: UVE: Page Tools Not Working with Traditional Rendering
### Problem Statement
Users are unable to utilize page tools analytics when using traditional rendering in dotCMS.

### Steps to Reproduce
1. Log in to dotCMS.
2. Edit a page in traditional mode
3. Try to use page tools analytics such as tracking and reporting.

### Acceptance Criteria
- Page tools analytics should be functional in both UVE and traditional rendering modes.
- Page Tools should have the `host_id` in the url to support multi site.
 
### External Links
N/A

### Assumptions & Initiation Needs
- It works in headless rendering

### Quality Assurance Notes & Workarounds
N/A

### PR DESCRIPTION
**Pull request by valentinogiardino**
### Proposed Changes
* Update URL construction logic to remove protocol and port for Mozilla tool.
* Adjust URL construction for Wave and Security Headers tools using `URL().origin` .

### Additional Info
This change aligns the URL formats with the requirements of the analytics tools, ensuring compatibility and preventing potential errors due to unsupported URL components. 

This PR fixes #29206

### Mozilla Observatory (Before and After)
image

This PR fixes: #29206

### CODE REVIEW

**Review by nicobytes — APPROVED** [review]
LGTM!

Just a thought, why in some tests use `any` when the function params are typed? Any edge case?

Screenshot 2024-08-20 at 8 58 20 AM

**Comment by valentinogiardino** [comment]
> LGTM!
> 
> Just a thought, why in some tests use `any` when the function params are typed? Any edge case?
> 
> Screenshot 2024-08-20 at 8 58 20 AM

Since `ellipsizeText` is a utility function, I aimed to ensure it handles edge cases and practices defensive programming. Even with TypeScript type enforcement, these tests provide an extra safety net and document how the function handles unexpected inputs. However, if you feel it's more appropriate, I can remove them.

File: core-web/libs/utils/src/lib/dot-utils.ts [hunk 1]
```diff
@@ -70,8 +70,13 @@ export function getRunnableLink(url: string, currentPageUrlParams: DotPageToolUr
     if (languageId) pageParams.append('language_id', String(languageId));
 
     // Replace placeholders in the base URL with actual values and append query parameters if they exist
+    const requestHostUrl = requestHostName ? new URL(requestHostName) : null;
     const finalUrl = url
-        .replace(/{requestHostName}/g, requestHostName ?? '')
+        .replace(
+            /{requestHostName}/g,
+            requestHostUrl ? `${requestHostUrl.protocol}//${requestHostUrl.hostname}` : ''
```
Comment (zJaaal) [inline core-web/libs/utils/src/lib/dot-utils.ts:77]: the URL object has a property that returns that. I think is `origin`

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/utils/src/lib/dot-utils.ts (hunk 1 above)
Comment (valentinogiardino) [reply core-web/libs/utils/src/lib/dot-utils.ts:77]: Nice

**Review by zJaaal — APPROVED** [review]
